lib/krane/template_sets.rb

```diff
@@ -134,11 +146,26 @@ def ejson_secrets_files
     end
 
     def validate
-      @template_sets.flat_map(&:validate)
+      errors = @template_sets.flat_map(&:validate)
+
+      if rendering_erb_disabled? && deploying_with_erb_files?
+        errors << "ERB template discovered with rendering disabled. If you were trying to render ERB and " \
+          "deploy the result, try piping the output of `krane render` to `krane-deploy` with the --stdin flag"
+      end
+
+      errors
     end
 
     private
 
+    def deploying_with_erb_files?
+      @template_sets.any?(&:deploying_with_erb_files?)
+    end
+
+    def rendering_erb_disabled?
+      !@template_sets.any?(&:render_erb)
+    end
+
```

Comment:
I feel the `TemplateSets` object should already know whether or not it allows its constituent template sets to render erb, it feels somewhat backwards to make it inspect them. Can you think of a case where we have some template sets that allow rendering while at the same time disallowing others?

Reply:
No not at all.

Reply:
In my experience when you have a collection class that owns an inner class you want one of them to be as stateless as possible, otherwise reasoning becomes really hard. We've picked the outer class to be the simple one so far. We definitely could have TemplateSets own `@render_erb` but it would be passing it to 2/4 of TemplateSet's public methods which feels a bit off.

Reply:
For a given template set, we can ask “does this have a @renderer?”. If no, it can’t render anything. Would this be a sufficient check?